text/XXX-template-poly.md

```diff
@@ -0,0 +1,154 @@
+- Title: Template poly redesign using sort poly
+
+- Drivers: Gaëtan Gilbert
+
+----
+
+# Summary
+
+Template poly currently provides 2 features in 1: implicit universe
+instances and adhoc sort polymorphism handling only Prop/Type.
+
+We want to make it less adhoc and base it on more well understood
+systems such as regular sort poly.
+
+This should also allow to stop producing useless constraints (like u <= prod.u0)
+when a template inductive is fully applied.
+
+# Detailed design
+
+## Declaration
+
+An inductive may be declared to use implicit univ instances when:
+
+- it is non mutual (nested is ok) (this rule is in master but maybe we will remove it)
+- its univ declaration has only univ variables unbounded from below
+- its univ declaration has at most 1 qvar variable.
```

Comment:
perhaps it would make sense to relax this condition, the extra qvars would appear in the parameters but not in the return sort. if we drop the above_prop condition it may be mildly useful, eg could define a template poly
~~~coq
Class foo A B := {}.
~~~
which can accept both SProp and Type for A and B